revolutionpi/digital_interrupts.go

```diff
@@ -98,11 +98,15 @@ func initializeDigitalInterrupt(pin SPIVariable, g *gpioChip, isEncoder bool) (*
 }
 
 func (di *diWrapper) Value(ctx context.Context, extra map[string]interface{}) (int64, error) {
-	return di.pin.Value()
+	val, err := di.pin.Value()
+	if err != nil {
+		return 0, err
+	}
+	return int64(val), nil
 }
 
 // Note: The revolution pi only supports uint32 counters, while the Value API expects int64.
-func (di *digitalInterrupt) Value() (int64, error) {
+func (di *digitalInterrupt) Value() (uint32, error) {
```

Comment:
With this change, `digitalInterrupt` no longer implements the `board.DigitalInterrupt` interface. I suspect you should change it back, but am open to disagreement.

If you keep it as-is (so it returns a uint32), update the comment on line 108.

Reply:
I think its fine because the diWrapper implements the interface

Reply:
might be worth renaming `digitalInterrupt` to avoid confusion for future readers though, is that something you guys would want?